core/src/main/scala/com/banno/kafka/consumer/ConsumerOps.scala

```diff
@@ -353,4 +353,85 @@ case class ConsumerOps[F[_], K, V](consumer: ConsumerApi[F, K, V]) {
       .recordStream(pollTimeout)
       .evalMap(r => process(r) <* consumer.commitSync(r.nextOffset))
 
+  /** Returns a stream that processes records using the specified function,
+    * committing offsets for successfully processed records, either after
+    * processing the specified number of records, or after the specified time
+    * has elapsed since the last offset commit. If the processing function
+    * returns a failure, the stream will halt with that failure, and the offsets
+    * will not be committed. This is at-least-once processing: records for which
+    * the function returns success, but whose offset have not yet been committed
+    * will be reprocessed after a subsequent failure. In some use cases this
+    * pattern is more appropriate than just using auto-offset-commits, since it
+    * will not commit offsets for failed records when the consumer is closed.
+    * The consumer must be configured to disable offset auto-commits.
+    */
+  def readProcessCommit2[A](
+      pollTimeout: FiniteDuration,
+      maxRecordCount: Long = 1000L,
+      maxCommitTime: Long = 60000L,
+  )(
+      process: ConsumerRecord[K, V] => F[A]
+  )(implicit C: Clock[F], S: Sync[F]): Stream[F, A] =
```

Comment:
Is the Sync needed? I couldn't find explicit usage below. If it is needed, maybe for a callee below that needs it, should Clock then be removed, since Sync implies Clock, and we can summon Clock without explicit reference?

Reply:
iirc `Ref.of` needed either `Sync` or `Concurrent`. I used `Sync` originally just to satisfy the compiler, but maybe `Concurrent` is better to use? I switched to that.